File: xtensus-frontend/src/app/models/personne.ts

import { Ville } from './ville';

export class Personne {
    id!: number;
    personneNom!: string;
    personnePrenom!: string;
    personneDateNaissance! : Date;
    personneMail! : string;
    numTel!: string;
    ville?: any; // Make ville optional and handle null values
gouvernorat?:any;
    [key: string]: any; // Index signature

    constructor() {
    }

  }

File: xtensus-backend/backend/src/main/java/tn/xtensus/backend/Entities/Personne.java
package tn.esprit.ihebsaidi.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults (level= AccessLevel.PRIVATE)
@Entity
public class Personne implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;
    String personneNom;
    String personnePrenom;
    Date personneDateNaissance;
    String personneMail;
    String numTel;

    @ManyToOne
    private Ville ville;




   

}
